src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,79 @@
+
+import com.hibob.academy.dao.*
+import jakarta.inject.Inject
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.springframework.stereotype.Component
+import java.sql.Date
+import kotlin.random.Random
+
+@Component
+
+class PetDao @Inject constructor(private val sql: DSLContext) {
+
+    private val pet = PetTable.instance
+    private val companyId = Random.nextLong()
+
+    private val petMapper = RecordMapper<Record, PetData> { record ->
+        PetData(
+            record[pet.name],
+            convertStringToPetType(record[pet.type]),
+            record[pet.companyId],
+            Date(record[pet.dateOfArrival].time)
+        )
+    }
```

Comment:
What about id?
Do we need to add id to the data class and the mapper?
Why dont you use the `dateOfArrival` as is?
Do we really need to convert it to `Date`

Reply:
why do i need to add the id?

for some reason if i dont convert it it doesnt work. date of arrival is of type Date